packages/backend/src/open_payments/invoice/model.ts

```diff
@@ -33,9 +35,73 @@ export class Invoice
 
   public processAt!: Date | null
 
-  public webhookAttempts!: number
-
   public get asset(): Asset {
     return this.account.asset
   }
+
+  public async onCredit({
+    balance,
+    createWithdrawal
+  }: onCreditOptions): Promise<void> {
+    if (balance >= this.amount) {
+      return await Invoice.transaction(async (trx) => {
+        await this.$query(trx).patch({
+          active: false
+        })
+        const event = await InvoiceEvent.query(trx).insertAndFetch({
+          type: InvoiceEventType.InvoicePaid,
+          data: this.toData(balance),
+          // TODO:
+          // Add 30 seconds to allow a prepared (but not yet fulfilled/rejected) packet to finish before being deactivated.
+          processAt: new Date()
+        })
+        const error = await createWithdrawal({
+          id: event.id,
+          account: this,
+          amount: balance,
+          timeout: BigInt(RETRY_LIMIT_MS) * BigInt(1e6) // ms -> ns
+        })
```

Comment:
As is, there's the possibility of the withdrawal transfer being created and the postgres transaction commit failing.
In which case, you'd need `onCredit` to be called again after the withdrawal times out or count on the balance to be withdrawn when the invoice expires. But there's a chance the invoice expiration handling happens before this withdrawal times out...
Alternatives include:
- the webhook service creates withdrawals on the first attempt to send the event
- every service with liquidity balance triggered webhooks (including asset, peer, open payment accounts) has workers handling accounts with recently updated balances, like outgoing payment and invoice services do/did.